src/server/nyplApiClient/index.ts

```diff
@@ -0,0 +1,128 @@
+import NyplApiClient from "@nypl/nypl-data-api-client"
+import aws from "aws-sdk"
+
+interface KMSCache {
+  clients: string[]
+  clientId: string
+  clientSecret: string
+}
+
+const config: any = {
+  api: {
+    platform: {
+      development:
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://qa-platform.nypl.org/api/v0.1",
+      production:
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://platform.nypl.org/api/v0.1",
+    },
+    // The 'discovery' base URL should use DISCOVERY_API_BASE_URL if set,
+    // falling back on PLATFORM_API_BASE_URL if set,
+    // and finally falling back on a sensible default.
+    discovery: {
+      development:
+        process.env.DISCOVERY_API_BASE_URL ||
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://qa-platform.nypl.org/api/v0.1",
+      production:
+        process.env.DISCOVERY_API_BASE_URL ||
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://platform.nypl.org/api/v0.1",
+    },
+    drbb: {
+      development:
+        process.env.DRB_API_BASE_URL || "http://drb-api-qa.nypl.org/search/",
+      production:
+        process.env.DRB_API_BASE_URL ||
+        "https://digital-research-books-api.nypl.org/v3/sfr/search",
+    },
+  },
+  tokenUrl: "https://isso.nypl.org/",
+}
+const appEnvironment = process.env.APP_ENV || "production"
+const kmsEnvironment = process.env.KMS_ENV || "encrypted"
+let decryptKMS
+let kms
+
+if (kmsEnvironment === "encrypted") {
+  kms = new aws.KMS({
+    region: "us-east-1",
+  })
+  // console.log({kms})
+
+  decryptKMS = async (key: string) => {
+    console.log("Decrypt func INPUT: " + key)
+    const params = {
+      CiphertextBlob: Buffer.from(key, "base64"),
+    }
+    const { Plaintext } = await kms.decrypt(params).promise()
+
+    return await new Promise((resolve, reject) => {
+      kms.decrypt(params, (err, data) => {
+        if (err) {
+          console.log({ err })
+          reject(err)
+        } else {
+          resolve(Plaintext.toString())
+        }
+      })
+    })
+  }
+}
+
+const clientId = process.env.clientId || process.env.PLATFORM_API_CLIENT_ID
```

Comment:
Why do we want to support both formats of clientId/PLATFORM_API_CLIENT_ID?

Reply:
Good catch, we can probably deprecate that.

Reply:
This was legacy. You're right and now would be a good chance to just use `PLATFORM_API_CLIENT_ID`.

Reply:
Just removed clientId.